Agents/FeatureInfoAgent/code/src/main/java/com/cmclinnovations/featureinfo/kg/TimeHandler.java

```diff
@@ -0,0 +1,386 @@
+
+            } else {
+                LOGGER.warn("No 'Measurement' or 'Forecast' property found, skipping this entry.");
+            }
+        }
+
+        try {
+            // Combine into a single instance (required by client before conversion to JSON can happen)
+            TimeSeries<Instant> combinedObject = Utils.getCombinedTimeSeries(tsClient, tsObjectList.values());
+
+            // Convert to JSON
+            JSONArray timeseriesJSON = tsClient.convertToJSON(
+                    new ArrayList<>(Arrays.asList(combinedObject)),
+                    Collections.nCopies(tsObjectList.size(), 1),
+                    new ArrayList<>(Arrays.asList(units)),
+                    new ArrayList<>(Arrays.asList(names))
+            );
+
+            // Add additionally reported parameters
+            JSONArray properties = new JSONArray();
+
+            for(int i = 0; i < measurements.length(); i++) {
+                JSONObject oldEntry = measurements.getJSONObject(i);
+                JSONObject newEntry = new JSONObject();
+
+                oldEntry.keySet().forEach(key ->  {
+                    if(!key.equals("Measurement") && !key.equals("Forecase") && !key.equals("Name") && !key.equals("Unit")) {
+                        newEntry.put(key, oldEntry.get(key));
+                    }
+                });
+
+                properties.put(newEntry);
+            }
+
+            // Add the properties
+            timeseriesJSON.getJSONObject(0).put("properties", properties);
+
+            // Return JSON
+            return timeseriesJSON;
+
+        } catch (Exception exception) {
+            LOGGER.error("Could not produce JSON objects from TimeSeries instances!", exception);
+
+            try {
+                response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+                response.getWriter().write("{\"description\":\"Could not produce JSON objects from TimeSeries instances!\"}");
+            } catch(IOException ioException) {
+                LOGGER.error("Could not write to HTTP response!", ioException);
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Builds a timeseries object for the input measurement IRI.
+     * 
+     * @param measureIRI IRI of measurement
+     * 
+     * @return timeseries object
+     */
+    protected TimeSeries<Instant> buildTimeseriesObject(String measureIRI) {
+
+        // Remove the brackets from the IRI as the timeseries client is shit and can't handle them
+        String fixedIRI = measureIRI;
+        if (fixedIRI.startsWith("<") && fixedIRI.endsWith(">")) {
+            fixedIRI = fixedIRI.substring(1, fixedIRI.length() - 1);
+        }
+
+        // Build then return the object
+        TimeSeries<Instant> result = null;
+        if(this.hours < 0) {
+            // Get all data
+            result = this.tsClient.getTimeSeries(new ArrayList<>(Arrays.asList(fixedIRI)));
```

Comment:
In the latest base lib, if you don't pass a connection object the these methods, it will open and close a connection for each query you make, if you want to make it faster, pass a connection object and close it at the end using try-with-resource

Reply:
This is something I can add in. Is there a method in the base lib to create the initial connection, or do I just use generic JDBC code?

Reply:
There's a RemoteRDBStoreClient.getConnection() method, which is basically calling the jdbc code